ir_ui_view_multi_company/models/ir_ui_view.py

```diff
@@ -0,0 +1,26 @@
+# Copyright 2025 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.osv.expression import AND
+
+
+class IrUiView(models.Model):
+    _inherit = "ir.ui.view"
+
+    company_id = fields.Many2one("res.company", string="Company", index=True)
+
+    @api.model
+    def _get_inheriting_views_domain(self):
+        domain = super()._get_inheriting_views_domain()
+        domain = domain if domain else []
+        return AND(
+            [
+                domain,
+                [
+                    "|",
+                    ("company_id", "=", False),
+                    ("company_id", "=", self.env.company.id),
```

Comment:
Wouldn't it be more convenient with `parent_of`?

```suggestion
                    ("company_id", "parent_of", self.env.company.id),
```

Reply:
I considered that, but I realized it could cause confusion if there's a specific view for the parent company and another for the subsidiary. By using '=', we ensure that only the views belonging to the exact company are selected